### PR TITLE
Move integration tests

### PR DESCRIPTION
**Pull request by megcrenshaw**
As part of the TeacherApplication test refactor, I moved the tests requiring `mount` to a separate file. I originally had these in the `integration` directory, but I had to import enzyme for it to work (see https://github.com/code-dot-org/code-dot-org/pull/45175/commits/14d5c239070f753e746c5e7b0cdac93a29c43d9e). Instead, I kept it in the `unit` directory but with a different name.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/test/unit/code-studio/pd/application/TeacherApplicationIntegrationTest.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    window.sessionStorage.getItem.restore();
+    window.sessionStorage.setItem.restore();
+    window.ga = undefined;
+  });
+
+  it('Sends firehose event on initialization and save', () => {
+    // Also calls firehose event on submit, but we reload the page on submit
+    // The page reload can't be stubbed because `window.location.reload`
+    // is not writable nor configurable: Object.getOwnPropertyDescriptor(window.location, 'toString')
+    // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty#modifying_a_property
```
Comment (megcrenshaw) [inline apps/test/unit/code-studio/pd/application/TeacherApplicationIntegrationTest.js:46]: I spent too long on this 😅 . We could do the window.location.reload in a separate function that we stub out, but then we'd have to export that function for the test here. I don't think there's a "good" way to test this––open to ideas

**Review by tim-dot-org — APPROVED** [review]
i think this is fine until we get a more holistic integration test approach. I don't think we'd want to run these separately (different karma/mocha config) anyways

File: apps/test/unit/code-studio/pd/application/TeacherApplicationIntegrationTest.js (hunk 1 above)
Comment (davidsbailey) [reply apps/test/unit/code-studio/pd/application/TeacherApplicationIntegrationTest.js:46]: so we actually do have a util already specifically for this purpose: https://github.com/code-dot-org/code-dot-org/blob/1274ce4088eb23433ba676c0871eb7950fd60b61/apps/src/utils.js#L725-L727 that is what I would recommend if you want to verify that a page reload has happened in a unit test.

**Comment by davidsbailey** [comment]
thank you for doing this Meg! This seems like a step in the right direction. it would be great for our engineering team to get on the same page about js integration tests.

File: apps/test/unit/code-studio/pd/application/TeacherApplicationIntegrationTest.js (hunk 1 above)
Comment (megcrenshaw) [reply apps/test/unit/code-studio/pd/application/TeacherApplicationIntegrationTest.js:46]: Ooo excellent––will update this!

**Pull request merged**